testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/TestClient.scala

```diff
@@ -488,15 +520,57 @@ class TestClient(
         workspaceBuildTargetsResult
       })
 
+  private def compareBuildTargetData(foundTarget: BuildTarget, target: BuildTarget): Boolean = {
+    List(Option(target.getData), Option(foundTarget.getData)) match {
+      case List(Some(targetData), Some(foundTargetData)) => targetData.getClass.isAssignableFrom(foundTargetData.getClass) &&
+        compareBuildTargetData(foundTargetData, targetData)
+      case List(None, None) => true
```

Comment:
You can match on a tuple instead of a `List`

```suggestion
    (Option(target.getData), Option(foundTarget.getData)) match {
      case (Some(targetData), Some(foundTargetData)) => targetData.getClass.isAssignableFrom(foundTargetData.getClass) &&
        compareBuildTargetData(foundTargetData, targetData)
      case (None, None) => true
```